src/posts/gpu-pipeline/index.md

```diff
@@ -0,0 +1,223 @@
+
+In this hackathon, we tried looking at different ways of reducing this bottleneck.
+
+### Data & Code Overview
+
+For this hackathon, we developed a benchmark of training a U-NET (with ResNet backend) model on the ERA-5 Dataset to predict next time steps. The U-Net model is implemented in PyTorch and the training pipeline is built using PyTorch DataLoader. The model can be trained on a single GPU or multiple GPUs using Distributed Data Parallel (DDP) for parallelization.
+
+-- TODO : Add an example image.
+
+The basic data loader is implemented in `zarr_ML_optimization/train_unet.py` and the model is defined in `zarr_ML_optimization/model.py`. The training pipeline is designed to be flexible and can be easily adapted to different datasets and models.
+
+More details on the model and training pipeline can be found in the [README](https://github.com/pangeo-data/ncar-hackathon-xarray-on-gpus/blob/main/zarr_ML_optimization/README.md) file in the `zarr_ML_optimization` folder.
+
+### Performance Bottlenecks
+
+First, we needed to identify the performance bottlenecks in our pipeline. We used NVIDIA's [Nsight Systems](https://developer.nvidia.com/nsight-systems) to profile our code and identify the areas that needed optimization.
+
+Here are some screenshots of the profiling results:
+
+- ![image](https://hackmd.io/_uploads/H1N1TTtAyl.png)
+- ![image](https://hackmd.io/_uploads/SyYA3pt0ye.png)
+
+The profiling results clearly showed that the data loading step was the main bottleneck in our pipeline. The time spent on data loading was significantly higher than the time spent on model training.
+
+This was also confirmed by a few other flags we added in our script to measure the time spent on data loading and model training. The results are shown below:
+
+![baseline plot](baseline.png)
+
+In the plot above, we show the throughput of the data loading and training steps in our pipeline. The three bars represent:
+
+- Real Data: Baseline throughput of the end-to-end pipeline using real data.
+- No Training (i.e. data loading throughput): Throughput of the data loading without any training (to measure the time spent on data loading vs. training).
+- Synthetic Data (i.e. Training throughput): Throughput of the data loading using synthetic data (to remove the data loading bottleneck).
+
+The results show that the data loading step is the main bottleneck in our pipeline, with **much** lower throughput compared to the training step.
+
+## Hackathon: Putting this altogether
+
+Our initial profiling showed that data loading is a major bottleneck in this workflow.
+
+During the hackathon, we tested the following strategies to improve the data loading performance:
+
+1. Optimized Chunking & Compression
+   - We explored different chunking and compression strategies to optimize the data loading performance. We found that using Zarr v3 with optimized chunking and compression significantly improved the data loading performance.
```

Comment:
I would either delete this summary or shorten and add a summary to the remaining items.

Some links to nvcomp and DALI would be great too